drivers/sensor/ite/ite_tach_it51xxx/tach_ite_it51xxx.c

```diff
@@ -0,0 +1,231 @@
+ * Tachometer Reading Register is still update based on the sampling counter
+ * that samples the tachometer input (T0A, T0B, T1A, T1B, T2A or T2B pins).
+ * The following is block diagram of this module:
+ *
+ *                                    Sample Rate = TACH_FREQ / 128
+ *                                                   |
+ *                            |        Tachometer 0  |                T0A (GPD6)
+ *                            |             |        | +-----------+   |
+ *                            |       +-----+-----+  | |   _   _   |<--+
+ *                            |------>|  T0L/MR   |<-+-|  | |_| |_ |<--+
+ *                            |       +-----------+    +-----------+   |
+ *                            |       capture pulses                  T0B (GPC6)
+ *                            |       in sample rate
+ *                            |       period
+ *                            |
+ *                            |        Sample Rate = TACH_FREQ / 128
+ *           +-----------+    |                      |
+ * Crystal-->| Prescaler |--->|        Tachometer 1  |                T1A (GPD7)
+ * 32.768k   +-----------+    |             |        | +-----------+   |
+ *                            |       +-----+-----+  | |   _   _   |<--+
+ *                            |------>|  T1L/MR   |<-+-|  | |_| |_ |<--+
+ *                            |       +-----------+    +-----------+   |
+ *                            |       capture pulses                  T1B (GPJ6)
+ *                            |       in sample rate
+ *                            |       period
+ *                            |
+ *                            |        Sample Rate = TACH_FREQ / 128
+ *                            |                      |
+ *                            |        Tachometer 2  |                T2A (GPJ0)
+ *                            |             |        | +-----------+   |
+ *                            |       +-----+-----+  | |   _   _   |<--+
+ *                            |------>|  T2L/MR   |<-+-|  | |_| |_ |<--+
+ *                            |       +-----------+    +-----------+   |
+ *                            |       capture pulses                  T2B (GPJ1)
+ *                            |       in sample rate
+ *                            |       period
+ *                            |
+ *
+ *
+ * Based on the counter value, we can compute the current RPM of external signal
+ * from encoders.
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/dt-bindings/sensor/it8xxx2_tach.h>
+#include <errno.h>
+#include <soc.h>
+#include <soc_dt.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(tach_ite_it51xxx, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * NOTE: The PWM output maximum is 324Hz in EC power saving mode, so if we need
+ *       fan to work, then don't let EC enter the mode.
+ */
```

Comment:
Is it more appropriate to put this NOTE in the pwm driver?

Reply:
Removed and it's already in pwm driver.